ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala

```diff
@@ -87,46 +87,66 @@ private class JdbcLedgerDao(
   ): Future[Option[ParticipantId]] =
-    dbDispatcher.executeSql(metrics.daml.index.db.getParticipantId)(storageBackend.participantId)
+    dbDispatcher
+      .executeSql(metrics.daml.index.db.getParticipantId)(storageBackend.ledgerIdentity)
+      .map(_.participantId)(DEC)
 
   /** Defaults to Offset.begin if ledger_end is unset
     */
   override def lookupLedgerEnd()(implicit loggingContext: LoggingContext): Future[Offset] =
-    dbDispatcher.executeSql(metrics.daml.index.db.getLedgerEnd)(storageBackend.ledgerEndOffset)
+    dbDispatcher
+      .executeSql(metrics.daml.index.db.getLedgerEnd)(storageBackend.ledgerEnd)
+      .map(_.lastOffset.getOrElse(Offset.beforeBegin))(DEC)
+
+  case class InvalidLedgerEnd(msg: String) extends RuntimeException(msg)
 
   override def lookupLedgerEndOffsetAndSequentialId()(implicit
       loggingContext: LoggingContext
   ): Future[(Offset, Long)] =
-    dbDispatcher.executeSql(metrics.daml.index.db.getLedgerEndOffsetAndSequentialId)(
-      storageBackend.ledgerEndOffsetAndSequentialId
-    )
+    dbDispatcher
+      .executeSql(metrics.daml.index.db.getLedgerEndOffsetAndSequentialId)(storageBackend.ledgerEnd)
+      .map { end =>
+        val EventSequentialIdBeforeBegin = 0L
+        (end.lastOffset, end.lastEventSeqId) match {
+          case (Some(offset), Some(seqId)) => (offset, seqId)
+          case (Some(offset), None) => (offset, EventSequentialIdBeforeBegin)
+          case (None, None) => (Offset.beforeBegin, EventSequentialIdBeforeBegin)
+          case (None, Some(seqId)) =>
+            throw InvalidLedgerEnd(
+              s"Parameters table in invalid state: ledger_end=None, ledger_end_sequential_id=$seqId"
+            )
+        }
+      }(DEC)
 
   override def lookupInitialLedgerEnd()(implicit
       loggingContext: LoggingContext
   ): Future[Option[Offset]] =
-    dbDispatcher.executeSql(metrics.daml.index.db.getInitialLedgerEnd)(
-      storageBackend.initialLedgerEnd
-    )
-
-  override def initializeLedger(
-      ledgerId: LedgerId
-  )(implicit loggingContext: LoggingContext): Future[Unit] =
-    dbDispatcher.executeSql(metrics.daml.index.db.initializeLedgerParameters) {
-      implicit connection =>
-        storageBackend.updateLedgerId(ledgerId.unwrap)(connection)
-    }
-
-  override def initializeParticipantId(
-      participantId: ParticipantId
+    dbDispatcher
+      .executeSql(metrics.daml.index.db.getInitialLedgerEnd)(storageBackend.ledgerEnd)
+      .map { end =>
+        end.lastOffset
+      }(DEC)
+
+  override def initialize(
+      ledgerId: LedgerId,
+      participantId: ParticipantId,
   )(implicit loggingContext: LoggingContext): Future[Unit] =
-    dbDispatcher.executeSql(metrics.daml.index.db.initializeParticipantId) { implicit connection =>
-      storageBackend.updateParticipantId(participantId.unwrap)(connection)
-    }
+    dbDispatcher
+      .executeSql(metrics.daml.index.db.initializeLedgerParameters) { implicit connection =>
```

Comment:
yo can omit the `{ implicit => connection` ... `(connection) }` and switch to `( ... )` instead.
(curry instead of implicit resolution)